digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -135,7 +135,7 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
         val ely: Option[Long] = roadLink.headOption.map(rl => MunicipalityDAO.getMunicipalityRoadMaintainers.getOrElse(rl.municipalityCode, 0))
         ely match {
           case Some(value) if value != 0 =>
-            Some(ReservedRoadPart(roadpartid, roadnumber, roadpart, lenght, Discontinuity.apply(discontinuity.toInt), value, Some(startDate), Some(endDate)))
+            Some(ReservedRoadPart(roadpartid, roadnumber, roadpart, lenght, Discontinuity.apply(discontinuity.toInt), value, Option(startDate), Option(endDate)))
```

Comment:
The change here was because the Option's apply function makes a null into None. When the end address is null from the DB, and if you do Some(null) = Some(null), and if you do Option(null)= None.
But i will check if i can change it in the DAO instead of in the Service class.